### PR TITLE
feat: update and fix publish.ts

### PR DESCRIPTION
**Pull request by jacoobes**

### CODE REVIEW

**Review by SrIzan10 — CHANGES_REQUESTED** [review]
i think you need to run prettier

File: TypeScript/publish.ts [hunk 1]
```diff
@@ -17,32 +17,25 @@
...
-	SlashCommand,
-} from "@sern/handler";
-import {
-	ApplicationCommandData,
-	ApplicationCommandType,
-	PermissionResolvable,
-} from "discord.js";
+import {CommandPlugin, CommandType, PluginType, SernOptionsData, SlashCommand,} from "@sern/handler";
+import {ApplicationCommandData, ApplicationCommandType, PermissionResolvable,} from "discord.js";
+import { useContainer } from "../index.js"; //your dependency getter
```
Comment (EvolutionX-10) [inline TypeScript/publish.ts]: explain pls

File: TypeScript/publish.ts [hunk 2]
```diff
@@ -55,99 +48,81 @@ export function publish(
...
-						? module.description
-						: undefined,
-					options: [CommandType.Slash, CommandType.Both].includes(
-						module.type
-					)
-						? optionsTransformer(
-								(module as SlashCommand).options ?? []
-						  )
-						: [],
+			const log = (...message : any[]) => () => console.log(...message);
```
Comment (EvolutionX-10) [inline TypeScript/publish.ts:58]: so lazy?

File: TypeScript/publish.ts [hunk 3]
```diff
@@ -55,99 +48,81 @@ export function publish(
...
+			const logged = (...message : any[]) => log(message);
```
Comment (EvolutionX-10) [inline TypeScript/publish.ts:59]: why tf we need this

File: TypeScript/publish.ts [hunk 4]
```diff
@@ -55,99 +48,81 @@ export function publish(
...
+			const appCmd = <V extends CommandType, S, T>(t: V) => {
+				return (is: S, els: T) => (t & CommandType.Both) !== 0
+					? is
+					: els
+			}
```
Comment (EvolutionX-10) [inline TypeScript/publish.ts:72]: add some comments explaining what it does, because eventually it'll be in someone's project

**Review by EvolutionX-10 — CHANGES_REQUESTED** [review]
about prettier, if you can run it, otherwise my workflow will automatically format it after this PR gets merged

File: TypeScript/publish.ts (hunk 3 above)
Comment (jacoobes) [reply TypeScript/publish.ts:59]: only write console log once, and can be interchanged with sern/logger easier

File: TypeScript/publish.ts (hunk 2 above)
Comment (jacoobes) [reply TypeScript/publish.ts:58]: kinda, its just to put it into callbacks

File: TypeScript/publish.ts (hunk 1 above)
Comment (jacoobes) [reply TypeScript/publish.ts]: ill explain more

**Review by EvolutionX-10 — APPROVED** [review]
# To be merged after Release of 2.0
- [ ] https://github.com/sern-handler/handler/pull/156

**Pull request merged**